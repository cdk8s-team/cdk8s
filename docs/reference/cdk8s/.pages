nav:
  - TypeScript: typescript.md
  - Python: python.md
  - Go: go.md