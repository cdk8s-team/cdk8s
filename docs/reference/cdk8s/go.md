# cdk8s (Go) <a name="API Reference"></a>

For Go API reference, please visit <https://pkg.go.dev/github.com/cdk8s-team/cdk8s-core-go/cdk8s/v2>.