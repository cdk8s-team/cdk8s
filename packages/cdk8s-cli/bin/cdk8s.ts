import * as yargs from 'yargs';

yargs
  .commandDir('cmds')
  .demandCommand()
  .help()
  .argv;
