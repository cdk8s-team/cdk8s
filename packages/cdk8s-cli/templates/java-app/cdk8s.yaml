language: java
app:  mvn exec:java -Dexec.mainClass="com.mycompany.app.Main"
imports:
  - k8s