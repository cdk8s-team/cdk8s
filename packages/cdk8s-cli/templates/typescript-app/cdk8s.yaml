language: typescript
app: node main.js
imports:
  - k8s
