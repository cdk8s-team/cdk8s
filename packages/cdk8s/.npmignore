# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
/.eslintrc.json
/.github
/.idea
/.projen
/.projenrc.js
/.versionrc.json
/.vscode
/coverage
/src
/test
/test-reports/
/tsconfig.eslint.json
/tsconfig.jest.json
/tsconfig.json
dist
junit.xml
!.jsii
!/lib
!/lib/**/*.d.ts
!/lib/**/*.js
