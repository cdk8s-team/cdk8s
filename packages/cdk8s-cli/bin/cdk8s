#!/usr/bin/env node
require('./cdk8s.js');