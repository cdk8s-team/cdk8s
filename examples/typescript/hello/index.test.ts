import { HelloKube } from './index';
import { Testing } from 'cdk8s';

test('snapshot', () => {
  const app = Testing.app();
  const chart = new HelloKube(app, 'hello');

  expect(Testing.synth(chart)).toMatchSnapshot();
});