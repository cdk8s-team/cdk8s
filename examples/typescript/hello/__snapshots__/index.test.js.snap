// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "hello-service-c8c17160",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "hello-k8s",
      },
      "type": "LoadBalancer",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "hello-deployment-c8c7fda7",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "hello-k8s",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "hello-k8s",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "paulbouwer/hello-kubernetes:1.7",
              "name": "hello-kubernetes",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
